pkg/builder/builder.go

```diff
@@ -120,7 +120,50 @@ func (b generalBuilder) NumGPUs() int {
 	return ir.NumGPUs()
 }
 
+// Always return updated when met error
+func (b generalBuilder) CheckDepsFileUpdate(ctx context.Context, tag string, deps []string) (bool, error) {
```

Comment:
Fancy implementation!

I am not sure how docker deals with the ImageCreateTime in our case. Did you test the logic locally?

Reply:
I test it. It seems that depends the tarball from the buildkit. So it could be cached if llb didn't change.